docs/topics/multiplatform/multiplatform-hierarchy.md

```diff
@@ -1,111 +1,322 @@
 [//]: # (title: Hierarchical project structure)
 
-With Kotlin 1.6.20, every new multiplatform project comes with a hierarchical project structure. This means that source
-sets form a hierarchy for sharing the common code among several targets. It opens up a variety of opportunities,
-including using platform-dependent libraries in common source sets and the ability to share code when creating multiplatform
-libraries.
-
-To get a default hierarchical project structure in your
-projects, [update to the latest release](releases.md#update-to-a-new-release). If you need to keep using an earlier version
-than 1.6.20, you can still enable this feature manually. For this, add the following to your `gradle.properties`:
-
-```none
-kotlin.mpp.enableGranularSourceSetsMetadata=true
-kotlin.native.enableDependencyPropagation=false
-```
+Multiplatform projects have support for hierarchical structure.
+This means you can form a hierarchy of source sets for sharing the common code among some, but not all,
+[supported targets](multiplatform-dsl-reference.md#targets).
+
+The Kotlin toolchain provides the correct default dependencies and locates the API surface area available in the shared code.
+This prevents cases like the use of a macOS-specific function in code shared for Windows. It opens up a variety of opportunities:
+
+* Using platform-dependent libraries in source sets shared among several native targets. One popular case is having access
+  to iOS-specific dependencies like `Foundation` when sharing code across all iOS targets.
+* Sharing code in similar targets, as well as publishing and consuming libraries with granular APIs targeting similar platforms.
+
+There are 3 ways to create a target hierarchy:
+
+* [Specify all targets and enable default hierarchy](multiplatform-hierarchy.md#default-hierarchy)
+* [Use target shortcuts available for typical cases](multiplatform-hierarchy.md#target-shortcuts)
+* [Manually declare and connect the source sets](multiplatform-hierarchy.md#manual-configuration).
+
+## Default hierarchy
+
+> The default target hierarchy is [Experimental](components-stability.md#stability-levels-explained). It may
+> be changed in future Kotlin releases without prior notice.
+> For the Kotlin Gradle build scripts, opt-in is required with `@OptIn(ExperimentalKotlinGradlePluginApi::class)`.
+>
+{type="warning"}
+
+Starting with Kotlin 1.8.20, you can set up source set hierarchy in your multiplatform projects with default target hierarchy.
+It's a template for all possible targets and their shared source sets hardcoded in the Kotlin Gradle plugin.
+
+When you declare the targets to which your project compiles,
+the plugin picks necessary source sets from the template and creates them in your project based on the specified targets.
+See the full hierarchy template:
```

Comment:
I think the full template interrupts the flow of reading heavily here. We just barely explained the basics, and most likely, it's not very clear what's going on without seeing an example, as we're avoiding strict terms and definitions and using "intuitive" wording like "the plugin picks necessary source sets" (which ones are necessary?), "based on the specific targets" (how exactly targets affect it?). 

Two options:
- Introduce a short example right here (you can then just reference it from the section with code sample, like: "This code will create the structure from the example above")
- Push full template somewhere below, and just give a hint to a reader that if they are confused at this point, it's OK and just keep reading, like: "Let's see how it works on the example"